validator/rpc/auth_token.go

```diff
@@ -136,53 +126,70 @@ func logValidatorWebAuth(validatorWebAddr, token string, tokenPath string) {
 			"the Prysm web interface",
 	)
 	log.Info(webAuthURL)
-	log.Infof("Validator CLient JWT for RPC and REST authentication set at:%s", tokenPath)
+	log.Infof("Validator Client Auth Token for RPC and REST authentication set at:%s", tokenPath)
 }
 
-func saveAuthToken(walletDirPath string, jwtKey []byte, token string) error {
-	hashFilePath := filepath.Join(walletDirPath, AuthTokenFileName)
+func saveAuthToken(tokenPath string, token string) error {
 	bytesBuf := new(bytes.Buffer)
-	if _, err := bytesBuf.WriteString(fmt.Sprintf("%x", jwtKey)); err != nil {
-		return err
-	}
-	if _, err := bytesBuf.WriteString("\n"); err != nil {
-		return err
-	}
 	if _, err := bytesBuf.WriteString(token); err != nil {
 		return err
 	}
 	if _, err := bytesBuf.WriteString("\n"); err != nil {
 		return err
 	}
 
-	if err := file.MkdirAll(walletDirPath); err != nil {
-		return errors.Wrapf(err, "could not create directory %s", walletDirPath)
+	if err := file.MkdirAll(filepath.Dir(tokenPath)); err != nil {
+		return errors.Wrapf(err, "could not create directory %s", filepath.Dir(tokenPath))
 	}
-
-	if err := file.WriteFile(hashFilePath, bytesBuf.Bytes()); err != nil {
-		return errors.Wrapf(err, "could not write to file %s", hashFilePath)
+	if err := file.WriteFile(tokenPath, bytesBuf.Bytes()); err != nil {
+		return errors.Wrapf(err, "could not write to file %s", tokenPath)
 	}
 
 	return nil
 }
 
-func readAuthTokenFile(r io.Reader) (secret []byte, token string, err error) {
-	br := bufio.NewReader(r)
-	var jwtKeyHex string
-	jwtKeyHex, err = br.ReadString('\n')
-	if err != nil {
-		return
+func readAuthTokenFile(r io.Reader) ([]byte, string, error) {
+	scanner := bufio.NewScanner(r)
+	var lines []string
+	var secret []byte
+	var token string
+	// Scan the file and collect lines, excluding empty lines
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.TrimSpace(line) != "" {
+			lines = append(lines, line)
+		}
 	}
-	secret, err = hex.DecodeString(strings.TrimSpace(jwtKeyHex))
-	if err != nil {
-		return
+
+	// Check for scanning errors
+	if err := scanner.Err(); err != nil {
+		return nil, "", err
 	}
-	tokenBytes, _, err := br.ReadLine()
-	if err != nil {
-		return
+
+	// Process based on the number of lines, excluding empty ones
+	switch len(lines) {
+	case 1:
+		// If there is only one line, interpret it as the token
+		token = strings.TrimSpace(lines[0])
+	case 2:
+		// TODO: Deprecate after a few releases
+		// For legacy files
+		// If there are two lines, the first is the jwt key and the second is the token
+		jwtKeyHex := strings.TrimSpace(lines[0])
+		s, err := hex.DecodeString(jwtKeyHex)
+		if err != nil {
+			return nil, "", errors.Wrapf(err, "could not decode JWT secret")
+		}
+		secret = bytesutil.SafeCopyBytes(s)
+		token = strings.TrimSpace(lines[1])
+		log.Warn("Auth token is a legacy file and should be regenerated.")
```

Comment:
```suggestion
		log.Warn("Auth token is a legacy file and should be regenerated.")
```
Do we have some documentation that we can link to here? Otherwise users not familiar with the update won't know what to do